New selinux type to restrict access rights.

AaronH88 opened the pull request.

gicmo commented:
> > Any thoughts on how to test this? I think our reverse dependency tests will verify that we didn't regress, but not sure how to verify that this has had the desired effect...?
> 
> Good question. Im not sure yet how to test this. I can see that one of the automated tests fail as I have the permissions incorrect. I need to figure out how to reproduce that

I was thinking of extending the `org.osbuild.test` stage or creating a new one to assert that itself is running in the correct domain and does not have certain capabilities. Add a new stage that is labelled with the privileged label and assert we have the correct label and do have the capabilities.

Example python code:

```python3
import selinux
import os
selinux.getpidcon(os.getpid())
[0, 'unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023']
```

Inspecting the capabilities could be done via `getpcaps` or `/proc/<pid>/status`.

teg commented on test/test_integration.py:
```diff
@@ -0,0 +1,25 @@
+        temp = subprocess.run(['osbuild', 'data/manifests/fedora-test-label-mac.json', '--output-directory=/tmp'],
+        capture_output=True, check=True)
+        output = str(temp.stdout)
+        output = re.split(',|:|\\\\n|\\\\t|\'|\\(b\'|None\\)' , output)
```
How is `output` meant to be used?

AaronH88 replied:
Yeah your right, It looks left over!

Ill remove it

AaronH88 replied:
removed :smile:

teg commented on test/test_integration.py:
```diff
@@ -0,0 +1,25 @@
+        temp = subprocess.run(['osbuild', 'data/manifests/fedora-test-label.json', '--output-directory=/tmp'],
+        capture_output=True, check=True)
+        output = str(temp.stdout)
+        output = re.split(',|:|\\\\n|\\\\t|\'|\\(b\'|None\\)' , output)
```
Same comment as above, maybe this is just a left-over (or I can't read ;) ).

teg commented on stages/org.osbuild.testmac:
```diff
@@ -0,0 +1,35 @@
+def main():
+
+    pid = '/proc/' + str(os.getpid()) + '/status'
+    temp = subprocess.Popen(['grep', 'CapBnd', pid], stdout = subprocess.PIPE)
```
Are we sure there will only be one match for this? Could other fields contain the same value? Might it make sense to iterate the lines of this file, parse out the fields, find the first line where the first field is `CapBnd` and use that? Just to be on the safe side.

AaronH88 replied:
Good plan to be safe.

The docs only mention the 5 `Cap` fields, but it could change in the future :smile:

AaronH88 replied:
implemented :smile:

teg commented on stages/org.osbuild.testmac:
```diff
@@ -0,0 +1,35 @@
+    temp = subprocess.Popen(['grep', 'CapBnd', pid], stdout = subprocess.PIPE)
+    # get the output as a string
+    output = str(temp.communicate())
+    output = re.split(',|:|\\\\n|\\\\t|\'|\\(b\'|None\\)' , output)
```
I think I know what this is trying to do, but my regex foo is not really able to verify it. Could you explain what it does in a comment (and why all these delimiters are included in the regex, is the format defined somewhere)?

AaronH88 replied:
Sure, ill add an inline comment!
FYI the escape characters look this way due to the format that subprocess.communicate() returns, its a lot of special characters that need to be escaped and the end result looks like my finger got stuck on the \ key :laughing:

AaronH88 replied:
I have added an inline comment, let me know if its not clear and I can reword :smile:

teg reviewed (changes requested):
Overall looking pretty good! Could you squash all this into one (or a couple if it makes sense) commits and explain what's going on in the commit messages?
A few comments inline.

AaronH88 commented:
> Overall looking pretty good! Could you squash all this into one (or a couple if it makes sense) commits and explain what's going on in the commit messages? A few comments inline.

Squashed them down to 5 commits using the style you linked me,
Also addressed the comments :smile:

teg reviewed (commented):
Style is looking great to me! And SELinux stuff too, to my untrained eye. I'll let @gicmo do the final approval/merge though.

de1987 commented:
the [PR 934](https://github.com/osbuild/osbuild/pull/934) is required for these changes

AaronH88 commented:
The epic is abandoned, closing this PR
